openslides_backend/action/actions/poll/stop.py

```diff
@@ -36,8 +38,98 @@ def update_instance(self, instance: Dict[str, Any]) -> Dict[str, Any]:
             self.control_countdown(meeting["poll_countdown_id"], "reset")
+
+        # calculate votescast, votesvalid, votesinvalid
+        voted_ids = poll.get("voted_ids", [])
+        instance["votescast"] = f"{len(voted_ids)}.000000"
+        if not meeting.get("users_enable_vote_weight") or not voted_ids:
+            instance["votesvalid"] = instance["votescast"]
+        else:
+            gmr = GetManyRequest(
+                Collection("user"), voted_ids, [f"vote_weight_${poll['meeting_id']}"]
+            )
+            gm_result = self.datastore.get_many([gmr])
+            users = gm_result.get(Collection("user"), {}).values()
+            instance["votesvalid"] = str(
+                sum(
+                    [
+                        Decimal(
+                            entry.get(f"vote_weight_${poll['meeting_id']}", "1.000000")
+                        )
+                        for entry in users
+                    ]
+                )
+            )
+        instance["votesinvalid"] = "0.000000"
+
+        # set entitled users at stop.
+        instance["entitled_users_at_stop"] = self.get_entitled_user_ids(poll)
         return instance
+
+    def get_entitled_user_ids(self, poll: Dict[str, Any]) -> List[Any]:
+        entitled_users = []
+        entitled_users_ids = set()
+        all_voted_users = poll.get("voted_ids", [])
+
+        # get all users from the groups.
+        meeting = self.datastore.get(
+            FullQualifiedId(Collection("meeting"), poll["meeting_id"]), ["group_ids"]
+        )
+        gmr = GetManyRequest(
+            Collection("group"), meeting.get("group_ids", []), ["user_ids"]
+        )
+        gm_result = self.datastore.get_many([gmr])
+        groups = gm_result.get(Collection("group"), {}).values()
+
+        for group in groups:
+            user_ids = group.get("user_ids", [])
+            if not user_ids:
+                continue
+            gmr2 = GetManyRequest(
+                Collection("user"),
+                list(user_ids),
+                [
+                    "id",
+                    "is_present_in_meeting_ids",
+                    f"vote_delegated_${poll['meeting_id']}_to_id",
+                ],
+            )
+            gm_result2 = self.datastore.get_many([gmr2])
+            users = gm_result2.get(Collection("user"), {}).values()
+            for user in users:
+                vote_delegated = {}
+                if user.get(f"vote_delegated_${poll['meeting_id']}_to_id"):
+                    vote_delegated = self.datastore.get(
+                        FullQualifiedId(
+                            Collection("user"),
+                            user[f"vote_delegated_${poll['meeting_id']}_to_id"],
+                        ),
+                        ["is_present_in_meeting_ids"],
+                    )
+
+                if user["id"] in entitled_users_ids:
+                    pass
```

Comment:
```suggestion
                    continue
```
may lead to confusing errors if someone reads it as `continue` and adds code at the end

Reply:
Done.